src/HealthChecks/HealthChecks/src/DefaultHealthCheckService.cs

```diff
@@ -29,12 +29,8 @@ internal sealed partial class DefaultHealthCheckService : HealthCheckService
         _scopeFactory = scopeFactory ?? throw new ArgumentNullException(nameof(scopeFactory));
         _options = options ?? throw new ArgumentNullException(nameof(options));
         _logger = logger ?? throw new ArgumentNullException(nameof(logger));
-
-        // We're specifically going out of our way to do this at startup time. We want to make sure you
-        // get any kind of health-check related error as early as possible. Waiting until someone
-        // actually tries to **run** health checks would be real baaaaad.
-        ValidateRegistrations(_options.Value.Registrations);
```

Comment:
Did this need to be moved? I think we should put it back here instead of in if not.

It's theoretically possible to use the `DefaultHealthCheckService` without the `HealthCheckPublisherHostedService` if you fish one out of the `IServiceCollection` or resolve it from a non-hosting `IServiceProvider`. And unlike `HealthCheckPublisherHostedService` which doesn't reference the registration names at all, `DefaultHealthCheckService` actually needs the names to be unique to build up `HealthReport.Entries`

Reply:
Updated - was a leftover from a previous attempt.